schema/brain/controller/plugins.py

```diff
@@ -215,6 +215,18 @@ def quick_change_desired_state(plugin_id, desired_state, conn=None):
         .run(conn)
 
 
+@wrap_rethink_errors
+@wrap_connection
+def recover_state(serv_name, conn=None):
+    curs = RPC.filter({"ServiceName":serv_name}).run(conn)
+    doc = curs.next()
```

Comment:
may throw a ValueError wrapping ReQL error if the cursor is empty.  May leave a hanging connection if there happens to be more than one document with the same serv_name.   Probably ok, but may consider interating the cursor to completion.